File: app/src/main/java/co/tpcreative/supersafe/ui/multiselects/adapter/CustomImageSelectAdapter.java
package co.tpcreative.supersafe.ui.multiselects.adapter;
import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;
import com.bumptech.glide.Glide;
import com.bumptech.glide.Priority;
import com.bumptech.glide.request.RequestOptions;

import java.util.ArrayList;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.services.SuperSafeApplication;
import co.tpcreative.supersafe.common.util.Utils;
import co.tpcreative.supersafe.model.EnumFormatType;
import co.tpcreative.supersafe.model.Image;
import co.tpcreative.supersafe.model.MimeTypeFile;
import co.tpcreative.supersafe.model.ThemeApp;

public class CustomImageSelectAdapter extends CustomGenericAdapter<Image> {
    private static final String TAG = CustomImageSelectAdapter.class.getSimpleName();
    private Context context;

    RequestOptions options = new RequestOptions()
            .centerCrop()
            .override(200 ,200)
            .placeholder(R.drawable.image_placeholder)
            .priority(Priority.HIGH);
    final ThemeApp themeApp = ThemeApp.getInstance().getThemeInfo();
    Drawable note1 = SuperSafeApplication.getInstance().getResources().getDrawable( themeApp.getAccentColor());

    public CustomImageSelectAdapter(Context context, ArrayList<Image> images) {
        super(context, images);
        this.context = context;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder viewHolder;
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.grid_view_item_image_select, null);
            viewHolder = new ViewHolder();
            viewHolder.imageView = (ImageView) convertView.findViewById(R.id.image_view_image_select);
            viewHolder.imgAudioVideo = (ImageView) convertView.findViewById(R.id.imgAudioVideo);
            viewHolder.tvTitle = (TextView) convertView.findViewById(R.id.tvTitle);
            viewHolder.view = convertView.findViewById(R.id.view_alpha);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }

        viewHolder.imageView.getLayoutParams().width = size;
        viewHolder.imageView.getLayoutParams().height = size;

        viewHolder.view.getLayoutParams().width = size;
        viewHolder.view.getLayoutParams().height = size;

        if (arrayList.get(position).isSelected) {
            viewHolder.view.setAlpha(0.5f);
            ((FrameLayout) convertView).setForeground(context.getResources().getDrawable(R.drawable.ic_done_white));

        } else {
            viewHolder.view.setAlpha(0.0f);
            ((FrameLayout) convertView).setForeground(null);
        }



        final Image data = arrayList.get(position);
        try {
            String extensionFile = Utils.getFileExtension(data.path);
            final MimeTypeFile mimeTypeFile = Utils.mediaTypeSupport().get(extensionFile);
            if (mimeTypeFile != null) {
                EnumFormatType formatTypeFile = EnumFormatType.values()[mimeTypeFile.formatType.ordinal()];
                switch (formatTypeFile){
                    case AUDIO:{
                        viewHolder.imgAudioVideo.setVisibility(View.VISIBLE);
                        viewHolder.imgAudioVideo.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_music_note_white_48));
                        viewHolder.tvTitle.setVisibility(View.VISIBLE);
                        viewHolder.tvTitle.setText(data.name);
                        Glide.with(context)
                                .load(note1)
                                .apply(options).into(viewHolder.imageView);
                        break;
                    }
                    case FILES:{
                        viewHolder.imgAudioVideo.setVisibility(View.VISIBLE);
                        viewHolder.imgAudioVideo.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_insert_drive_file_white_48));
                        viewHolder.tvTitle.setVisibility(View.VISIBLE);
                        viewHolder.tvTitle.setText(data.name);
                        Glide.with(context)
                                .load(note1)
                                .apply(options).into(viewHolder.imageView);
                        break;
                    }
                    case VIDEO:{
                        viewHolder.imgAudioVideo.setVisibility(View.VISIBLE);
                        viewHolder.imgAudioVideo.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_videocam_white_36));
                        viewHolder.tvTitle.setVisibility(View.INVISIBLE);;
                        Glide.with(context)
                                .load(data.path)
                                .apply(options).into(viewHolder.imageView);
                        break;
                    }
                    default:{
                        viewHolder.tvTitle.setVisibility(View.INVISIBLE);
                        viewHolder.imgAudioVideo.setVisibility(View.INVISIBLE);
                        Glide.with(context)
                                .load(data.path)
                                .apply(options).into(viewHolder.imageView);
                        break;
                    }
                }
            }
            else {
                viewHolder.tvTitle.setVisibility(View.INVISIBLE);
                Glide.with(context)
                        .load(R.drawable.ic_music)
                        .apply(options).into(viewHolder.imageView);
            }
        }
        catch (Exception e){
            e.printStackTrace();
        }
        return convertView;
    }

    private static class ViewHolder {
        public ImageView imageView;
        public ImageView imgAudioVideo;
        public View view;
        public TextView tvTitle;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/ui/multiselects/adapter/CustomImageSelectAdapter.kt
package co.tpcreative.supersafe.ui.multiselects.adapter
import android.content.Context
import android.graphics.drawable.Drawable
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.common.services.SuperSafeApplication
import co.tpcreative.supersafe.common.util.Utils
import co.tpcreative.supersafe.model.EnumFormatType
import co.tpcreative.supersafe.model.ImageModel
import co.tpcreative.supersafe.model.MimeTypeFile
import co.tpcreative.supersafe.model.ThemeApp
import com.bumptech.glide.Glide
import com.bumptech.glide.Priority
import com.bumptech.glide.request.RequestOptions
import kotlinx.android.synthetic.main.grid_view_item_image_select.view.*
import java.util.*

class CustomImageSelectAdapter(context: Context?, images: ArrayList<ImageModel>?) : CustomGenericAdapter<ImageModel>(context, images) {
    var options: RequestOptions? = RequestOptions()
            .centerCrop()
            .override(200, 200)
            .placeholder(R.drawable.image_placeholder)
            .priority(Priority.HIGH)
    val themeApp: ThemeApp? = ThemeApp.getInstance()?.getThemeInfo()
    var note1: Drawable? = ContextCompat.getDrawable(SuperSafeApplication.getInstance(),themeApp?.getAccentColor()!!)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var mConvertView = convertView
        val viewHolder: ViewHolder
        if (mConvertView == null) {
            mConvertView = layoutInflater?.inflate(R.layout.grid_view_item_image_select, null)
            viewHolder = ViewHolder()
            viewHolder.imageView = mConvertView?.image_view_image_select
            viewHolder.imgAudioVideo = mConvertView?.imgAudioVideo
            viewHolder.tvTitle = mConvertView?.tvTitle
            viewHolder.view = mConvertView?.view_alpha
            mConvertView?.setTag(viewHolder)
        } else {
            viewHolder = mConvertView.tag as ViewHolder
        }
        viewHolder.imageView?.layoutParams?.width = size
        viewHolder.imageView?.layoutParams?.height = size
        viewHolder.view?.layoutParams?.width = size
        viewHolder.view?.layoutParams?.height = size
        if (arrayList?.get(position)?.isSelected!!) {
            viewHolder.view?.alpha = 0.5f
            (mConvertView as FrameLayout?)?.setForeground(ContextCompat.getDrawable(context!!,R.drawable.ic_done_white))
        } else {
            viewHolder.view?.alpha = 0.0f
            (mConvertView as FrameLayout?)?.foreground = null
        }
        val data = arrayList!![position]
        try {
            val extensionFile: String? = Utils.getFileExtension(data.path)
            val mimeTypeFile: MimeTypeFile? = Utils.mediaTypeSupport().get(extensionFile)
            if (mimeTypeFile != null) {
                when (EnumFormatType.values()[mimeTypeFile.formatType?.ordinal!!]) {
                    EnumFormatType.AUDIO -> {
                        viewHolder.imgAudioVideo?.visibility = View.VISIBLE
                        viewHolder.imgAudioVideo?.setImageDrawable(ContextCompat.getDrawable(context!!,R.drawable.baseline_music_note_white_48))
                        viewHolder.tvTitle?.visibility = View.VISIBLE
                        viewHolder.tvTitle?.text = data.name
                        Glide.with(context!!)
                                .load(note1)
                                .apply(options!!).into(viewHolder.imageView!!)
                    }
                    EnumFormatType.FILES -> {
                        viewHolder.imgAudioVideo?.visibility = View.VISIBLE
                        viewHolder.imgAudioVideo?.setImageDrawable(ContextCompat.getDrawable(context!!,R.drawable.baseline_insert_drive_file_white_48))
                        viewHolder.tvTitle?.visibility = View.VISIBLE
                        viewHolder.tvTitle?.text = data.name
                        Glide.with(context!!)
                                .load(note1)
                                .apply(options!!).into(viewHolder.imageView!!)
                    }
                    EnumFormatType.VIDEO -> {
                        viewHolder.imgAudioVideo?.visibility = View.VISIBLE
                        viewHolder.imgAudioVideo?.setImageDrawable(ContextCompat.getDrawable(context!!,R.drawable.baseline_videocam_white_36))
                        viewHolder.tvTitle?.visibility = View.INVISIBLE
                        Glide.with(context!!)
                                .load(data.path)
                                .apply(options!!).into(viewHolder.imageView!!)
                    }
                    else -> {
                        viewHolder.tvTitle?.visibility = View.INVISIBLE
                        viewHolder.imgAudioVideo?.visibility = View.INVISIBLE
                        Glide.with(context!!)
                                .load(data.path)
                                .apply(options!!).into(viewHolder.imageView!!)
                        Utils.Log(TAG,"File select path  ${data.path}")
                    }
                }
            } else {
                viewHolder.tvTitle?.visibility = View.INVISIBLE
                Glide.with(context!!)
                        .load(R.drawable.ic_music)
                        .apply(options!!).into(viewHolder.imageView!!)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return mConvertView!!
    }

    private class ViewHolder {
        var imageView: AppCompatImageView? = null
        var imgAudioVideo: AppCompatImageView? = null
        var view: View? = null
        var tvTitle: AppCompatTextView? = null
    }

    companion object {
        private val TAG = CustomImageSelectAdapter::class.java.simpleName
    }

    init {
        this.context = context
    }
}